modes/elixir-mode/run-elixir.lisp

```diff
@@ -0,0 +1,86 @@
+(defpackage :lem-elixir-mode.run-elixir
+  (:use :cl :lem :lem-elixir-mode)
+  (:export :*elixir-run-command*
+           :run-elixir))
+
+(in-package :lem-elixir-mode.run-elixir)
+
+(defvar *elixir-run-command* "iex")
+
+(defvar *process* nil)
+
+(define-major-mode run-elixir-mode lem-elixir-mode:elixir-mode
+    (:name "elixir"
+     :keymap *run-elixir-mode-keymap*
+     :syntax-table lem-elixir-mode::*elixir-syntax-table*)
+  (reset-listener-variables (current-buffer))
+  (lem/listener-mode:start-listener-mode))
+
+(define-key lem-elixir-mode::*elixir-mode-keymap* "C-c C-r" 'elixir-eval-region)
+
+(defun reset-listener-variables (buffer)
+  (setf (variable-value 'lem/listener-mode:listener-set-prompt-function :buffer buffer)
+        #'identity
+        (variable-value 'lem/listener-mode:listener-check-input-function :buffer buffer)
+        (constantly t)
+        (variable-value 'lem/listener-mode:listener-execute-function :buffer buffer)
+        'execute-input))
+
+(defun execute-input (point string)
+  (declare (ignore point))
+  (unless (alive-process-p)
+    (editor-error "Elixir process doesn't exist."))
+  (lem-process:process-send-input *process*
+                                  (concatenate 'string string (string #\newline))))
+
+(defun alive-process-p ()
+  (and *process*
+       (lem-process:process-alive-p *process*)))
+
+(defun repl-buffer-exists-p ()
+  (get-buffer "*elixir*"))
+
+(defun get-repl-buffer ()
+  (let ((buffer (make-buffer "*elixir*")))
+    (unless (eq (buffer-major-mode buffer) 'run-elixir-mode)
+      (change-buffer-mode buffer 'run-elixir-mode))
+    buffer))
+
+(defun output-callback (string)
+  (let* ((already-exists (repl-buffer-exists-p))
+         (buffer (get-repl-buffer))
+         (p (buffer-point buffer)))
+    (buffer-end p)
+    (setf string (ppcre:regex-replace-all "\\r\\n" string (string #\newline)))
+    (insert-string p string)
+    (when (ppcre:scan "^(iex|\.\.\.)\(.+\)" (line-string p))
```

Comment:
Doesn't the backslash inside a string literal in Common Lisp disappear if there is only one?
```
"\." => "."
```

Reply:
Maybe? I'm not an expert in regex, I converted the Emacs regex, the Emacs equivalent is:
"^\\(iex\\|\\.\\.\\.\\)(.+)>"

The otuput comes from the command rxt-elisp-to-pcre in pcre2el

Reply:
In ppcre, no backslash seems to be correct.

```
CL-USER> (ppcre:parse-string ".\\.")
(:SEQUENCE :EVERYTHING #\.)
```

Reply:
Oh, interesting, good to know :+1:  (I modify the regex, to remove the backslash)